### PR TITLE
Feature/warnings and errors

### LINKED ISSUE #136: Compilation error for ignored class
I have an input file that contains two protocols and one class.  I would like Cuckoo to ignore the class.

Despite running cuckoo with the options `--no-classes` and `--exclude ClassName` I still get errors from Cuckoo trying to mock the class.  The class contains private variables with inferred types (e.g. `private let foo = true`) and Cuckoo gives the error:

```
error: Type of instance variable frames could not be inferred. Please specify it explicitly. (/path/to/input.swift)
```

I would expect that Cuckoo doesn't even do these checks if it's told to ignore this class.

### PR DESCRIPTION
**Pull request by MatyasKriz**
Resolves #136

### CODE REVIEW

File: Generator/Source/CuckooGeneratorFramework/Tokenizer.swift [hunk 1]
```diff
@@ -210,7 +240,7 @@ public struct Tokenizer {
         }
```
Comment (MatyasKriz) [inline Generator/Source/CuckooGeneratorFramework/Tokenizer.swift]: 2 lines above this we still print to `stderr`. I couldn't find a way to propagate the error up to the generator.

File: Generator/Source/CuckooGeneratorFramework/Tokenizer.swift [hunk 2]
```diff
@@ -9,6 +9,29 @@
 import Foundation
 import SourceKittenFramework
 
+public struct TokenizationResult {
+    public let token: Token?
+    private(set) public var errors = [] as [TokenizationError]
```
Comment (TadeasKriz) [inline Generator/Source/CuckooGeneratorFramework/Tokenizer.swift]: `public let`

File: Generator/Source/CuckooGeneratorFramework/Tokenizer.swift [hunk 3]
```diff
@@ -9,6 +9,29 @@
 import Foundation
 import SourceKittenFramework
 
+public struct TokenizationResult {
```
Comment (TadeasKriz) [inline Generator/Source/CuckooGeneratorFramework/Tokenizer.swift]:
```
public struct TokenizationResult<TOKEN: Token> {
    public let token: TOKEN?
    public let errors: [TokenizationError]

    public init(token: TOKEN?, errors: [TokenizationError]) {
        self.token = token
        self.errors = errors
    }
}
```

File: Generator/Source/CuckooGeneratorFramework/Tokenizer.swift [hunk 4]
```diff
@@ -22,54 +45,58 @@ public struct Tokenizer {
...
+        let representables: [SourceKitRepresentable] = structure.dictionary[Key.substructure.rawValue] as? [SourceKitRepresentable] ?? []
+        let declarations = tokenize(representables)
+        let imports = tokenize(imports: declarations.flatMap { $0.token })
 
         return FileRepresentation(sourceFile: file, declarations: declarations + imports)
     }
 
-    private func tokenize(_ representables: [SourceKitRepresentable]) -> [Token] {
-        return representables.flatMap(tokenize)
+    private func tokenize(_ representables: [SourceKitRepresentable]) -> [TokenizationResult] {
```
Comment (TadeasKriz) [inline Generator/Source/CuckooGeneratorFramework/Tokenizer.swift]: `[TokenizationResult<Token>]`

File: Generator/Source/CuckooGeneratorFramework/Tokenizer.swift [hunk 5]
```diff
@@ -22,54 +45,58 @@ public struct Tokenizer {
...
+        return representables.flatMap(tokenize(_:))
     }
 
     internal static let nameNotSet = "name not set"
     internal static let unknownType = "unknown type"
-    private func tokenize(_ representable: SourceKitRepresentable) -> Token? {
+    private func tokenize(_ representable: SourceKitRepresentable) -> TokenizationResult? {
```
Comment (TadeasKriz) [inline Generator/Source/CuckooGeneratorFramework/Tokenizer.swift]: `TokenizationResult<Token>`

File: Generator/Source/CuckooGeneratorFramework/Tokenizer.swift [hunk 6]
```diff
@@ -171,6 +199,8 @@ public struct Tokenizer {
             // stderrPrint("Unknown kind. Dictionary: \(dictionary) \(file.path ?? "")")
             return nil
         }
+
+        return TokenizationResult(token: token, errors: errors)
     }
 
     private func tokenize(methodName: String, parameters: [SourceKitRepresentable]) -> [MethodParameter] {
```
Comment (TadeasKriz) [inline Generator/Source/CuckooGeneratorFramework/Tokenizer.swift]: `[TokenizationResult<MethodParameter>]`

File: Generator/Source/CuckooGeneratorFramework/Tokenizer.swift [hunk 7]
```diff
@@ -182,23 +212,23 @@ public struct Tokenizer {
...
+            let kind = dictionary?[Key.kind.rawValue] as? String ?? ""
             return kind == Kinds.MethodParameter.rawValue
-        })
+        }
 
-        return zip(parameterLabels, filteredParameters).flatMap(tokenize)
+        return zip(parameterLabels, filteredParameters).flatMap(tokenize(parameterLabel:parameter:))
     }
 
     private func tokenize(parameterLabel: String?, parameter: SourceKitRepresentable) -> MethodParameter? {
```
Comment (TadeasKriz) [inline Generator/Source/CuckooGeneratorFramework/Tokenizer.swift]: `TokenizationResult<MethodParameter>`

File: Generator/Source/CuckooGeneratorFramework/Tokenizer.swift (hunk 1 above)
Comment (TadeasKriz) [reply Generator/Source/CuckooGeneratorFramework/Tokenizer.swift]: `return TokenizationResult(token: nil, errors: [.warning(" .... ")])`

**Comment by MatyasKriz** [comment]
I'll close this outdated PR for now.

**Pull request closed without merging**